engine/client_internal.go

```diff
@@ -127,11 +128,28 @@ func (c *Client) GetWebhooks(ctx context.Context) ([]notifications.ModelWebhook,
 	return c.options.notifications.webhookManager.GetAll(ctx)
 }
 
-// loadPaymailClient will load the Paymail client
-func (c *Client) loadPaymailClient() (err error) {
+// loadPaymailComponents will load the Paymail client
+func (c *Client) loadPaymailComponents() (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			if e, ok := r.(error); ok {
+				err = spverrors.Wrapf(e, "error when creating paymail components")
+			} else {
+				err = spverrors.Newf("error when creating paymail components: %v", r)
+			}
```

Comment:
```suggestion
            finalErr := r.(error) ? err : spverrors.Newf("error when creating paymail components: %v", r)
            err = spverrors.ErrorWhenCreatingPaymailComponents.Wrap()
```

Of course there is no ternary operator available in go xD

Reply:
in other words, in case if r is not error, then I will wrap "error when creating paymail components: %v" into "error when creating paymail components", doesn't seem to be a better solution

Reply:
Maybe I didn't mention about a key difference - 
on the caller side, my solution will allow to do this:
```
if errors.Is(err, spverrors.ErrorWhenCreatingPaymailComponents) {
 // do sth
}
```

Reply:
does it really matter in this case? I mean, from my point of view it should be just panic, but I'm adjusting it to current solution, where engine is eventually returning an error on creation, that's why it's this way.
For me it doesn't matter what is this error, because something go wrong in instantiation of whole engine, and actually whole application should just stop.

Reply:
I get it ;) 
Consider this comment resolved 👍